python/ray/serve/_private/autoscaling_policy.py

```diff
@@ -202,3 +227,54 @@ def get_decision_num_replicas(
             self.decision_counter = 0
 
         return decision_num_replicas
+
+    def set_target_capacity(
+        self,
+        new_target_capacity: Optional[float],
+        new_target_capacity_direction: Optional[TargetCapacityDirection],
+    ):
+        self.target_capacity = new_target_capacity
+        self.target_capacity_direction = new_target_capacity_direction
```

Comment:
is there a reason this needs to be made stateful rather than passing it into the `calculate_desired_num_replicas` call?

Reply:
I see, it's also used in `get_current_lower_bound`

Reply:
Yes, but all these methods can be made stateless. I'm working on that right now.

Reply:
I made it stateless.